mantidproject/mantid
# Add options dialog to Reflectometry GUI

**Description of work.**
Added options dialog to the reflectometry tools menu which allows control of warnings in #26533.

**To test:**
- Open ISISReflectometry GUI
- Tools -> Options
- Select some options
- Close the dialog
- Reopen the dialog (observe persistent options)
- Repeat with closing the GUI/Mantid
- Ensure options are saved/loaded correctly
- Check that options are discarded on cancel and close (x)

Check warnings and round precision are behaving as expected.

**Note:**
Possible future improvements include :
- Save on Mantid/GUI/batch close
- Avoid warning if setting changed and changed back
- Make round precision work with `ReductionJobs` 'model'
- Disable toolbar when processing (observe crash if load attempted)

Closes #26533 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
